ExecutorPluginNG-ERP-4.0/SimulationPlugin.cs

```diff
@@ -0,0 +1,38 @@
+﻿using DistributedExperimentation.DataModel;
+using DistributedExperimentation.Experimenter.ExecutorPlugin;
+using System;
+using System.Collections.Generic;
+
+namespace ExecutorPluginNGERP40
+{
+    public class ExecutorPlugin : IExecutorPlugin
+    {
+        public void execute(IExperimentSeries experimentSeries)
+        {
+            var experimentCollection = experimentSeries.getExperiments();
+            foreach (var experiment in experimentCollection)
+            {
+                SolveCollection((IParameterList)experiment.getParameters());
+            }
+        }
+
+        private void SolveCollection(IParameterList parameterList)
+        {
+            for (uint i = 0; i < parameterList.count(); i++)
+            {
+                var parameter = parameterList.get(i);
+                if (parameter.getValue().isPrimitive())
+                {
+                    System.Diagnostics.Debug.WriteLine("Parameter: {0}", i);
+                    System.Diagnostics.Debug.WriteLine(parameter.getValue());
```

Comment:
parameter.getValue() somehow returns the type not the value. that should be fixed.

Reply:
I don't really understand what you mean or what you expected. getValue() method of an IParameter gets in your case an object instance of an primitive data type, which is defined in the space of DataModel as an type of BooleanValue, CharacterStringValue, IntegerValue or RealValue. To get the primitive value as it defined in .Net call the specific method of one of these types i described above. 

For example:
`if (parameter.getValue().isPrimitive())
 {
    parameterValue = parameter.getValue();
    // value representation as string
    String rawValue = parameterValue.getRawValue(); 
    if (parameterValue.isInteger())
    {
       // value representation as its interpreted type in .Net
       int intValue = parameterValue.getIntegerValue();
    }
}`

This is necessary because in .Net is a string type a complex data type (not an inheritance of ValueType). I will described this also in my thesis.

Reply:
THIS:
if (param.getValue().isPrimitive())
{
   Debug.WriteLine("Parameter No.: {0}", i);
   Debug.WriteLine("parameter.getValue() -> {0}", parameter.getValue());
   Debug.WriteLine("parameter.getValue().GetType() -> {0}", parameter.getValue().GetType());
}

Results in: 

Parameter No.: 0
param.getValue() -> DistributedExperimentation.DataModel.Implementation.IntegerValue
param.getValue().GetType() -> DistributedExperimentation.DataModel.Implementation.IntegerValue

there is no method on param for getRawValue() do i miss a cast ? even then it is no good solution.

Reply:
```
var p = parameter.getValue() as IPrimitiveValue;
Debug.WriteLine("parameter -> {0}", p.getRawValue());
```

you need an upcast, but it is still not a good solution. may move getRawValue() method to IParameterValue? even then the usual user would expect to get the RawValue out of GetValue() like everywhere else.